server/server.go

```diff
@@ -26,27 +26,40 @@ import (
 )
 
 // New returns an AdapterServer that wraps the given high-level
-// Adapter implementation with the Tokens field populated.
-// The stop channel is used to signal when the file watcher should
+// Adapter implementation with the Tokens field populated from the file
+// which name is configured in the AUTH_TOKENS_PATH environment variable.
+// The stop channel returned is used to signal when the file watcher should
 // be closed and stop watching for file changes.
-func New[Config any](adapters map[string]framework.Adapter[Config], stop <-chan struct{}) api_adapter_v1.AdapterServer {
+func New[Config any](
+	adapters map[string]framework.Adapter[Config],
+) (api_adapter_v1.AdapterServer, chan struct{}) {
```

Comment:
I originally had the `stop` channel listed as a param, but I think it's a better UX to create this channel internally and return it. Lmk if you prefer the first approach and I can revert

Reply:
Think of it like a Context.